Add class file to get source packages.

Yoshihiro-Okada-ML opened the pull request:
# Purpose of pull request

This class downloads source code and debian related files for all packages which are included in rootfs.

# Test
## How to test

Add following line into the local.conf.

```
INHERIT += "dl-srcpkg"
```

And then type following build command.

```
$ bitbake emlinux-image-base
```

## Test result

All source code and debian related packages are located under the following directory.

```
tmp/deploy/sources/xxxxx
```

Yoshihiro-Okada-ML committed:
Add class file to get source packages.

This class downloads source code and debian related files for all
packages which are included in rootfs.

masami256 commented on classes/dl-srcpkg.bbclass:
```diff
@@ -0,0 +1,57 @@
+    ### setup network & apt environment
+    sudo cp /etc/resolv.conf '${ROOTFSDIR}'/etc/resolv.conf
+    prepare_local_isarapt
+    sudo sh -c "echo 'deb [trusted=yes] copy:///tmp/isar-apt/emlinux-bookworm-arm64/apt/emlinux-bookworm isar main' > '${ROOTFSDIR}'/etc/apt/sources.list.d/isar-apt.list"
```
please fix hardcoded architecture name in line 38 and 39.

Yoshihiro-Okada-ML replied:
I replaced some hardcoded strings and omitted some needless quote.

masami256 replied:
thanks.

masami256 commented on classes/dl-srcpkg.bbclass:
```diff
@@ -0,0 +1,59 @@
+# INHERIT += "dl-srcpkg"
+#
+
+LOCAL_APT_DIR="tmp/isar-apt/${DISTRO}-${DISTRO_ARCH}/apt/${DISTRO}"
```
tmp directory is changeable so ${TMPDIR} would be better.

Yoshihiro-Okada-ML replied:
TMPDIR is a absolute path from "/" on HOST PC environment. This LOCAL_APT_DIR is relative directory in rootfs environment. So we can't use TMPDIR.

masami256 commented:
> All source code and debian related packages are located under the following directory.
> tmp/deploy/images/xxxxx/sources

How about put files one of following directory instead of tmp/deploy/images/<MACHINE>/sources
- tmp/deploy/sources
- tmp/deploy/<MACHINE>/sources
- tmp/deploy/sources/<MACHINE>

Yoshihiro-Okada-ML committed:
Check /etc/resolv.conf exist or not.

Some package may create their own /etc/resolv.conf.
So check this before coping /etc/resolv.conf from build host.

masami256 commented:
LGTM

hiraku-wfs commented:
LGTM

masami256 commented:
@Yoshihiro-Okada-ML thanks.
